src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -355,6 +392,11 @@ public HdAccount GetUnusedAccount(string walletName, string password)
 
             Wallet wallet = this.GetWalletByName(walletName);
 
+            if (wallet.IsExtPubKeyWallet)
+            {
+                throw new CannotAddAccountToXpubKeyWalletException("Use recover-via-extpubkey instead.");
```

Comment:
Excellent.
Maybe just throw a `WalletException` here (ref my previous comment).

Reply:
as per above, I wouldn't want to catch and then interrogate the message when building the error response - what do you think? Seems to be logical to throw the explicit as it is caught and the walletexception is the one that goes to the client caller

Reply:
ah it's because you want to return a HTTP (403) Forbidden. Fair enough.